nevergrad/optimization/base.py

```diff
@@ -439,6 +440,17 @@ def ask(self) -> p.Parameter:
             else:
                 candidate = self._internal_ask_candidate()
                 # only register actual asked points
+
+            # In the deterministic case, possibly mutate.
+            num_determinism_failures = 0
+            while candidate.get_value_hash() in self._forbidden_value:
+                num_determinism_failures += 1
+                if num_determinism_failures >= 10:
```

Comment:
with this limit we get rid of the risk of infinite / super-long loop.

Still, it would be better to have the auxiliary optimizer (which ensures that we do not go too far) - but Roma was not built in a single day.